### PR TITLE
remove stack trace and update submodule

### PR DESCRIPTION
**Pull request by diosmosis**

### CODE REVIEW

File: core/ArchiveProcessor/PluginsArchiver.php [hunk 1]
```diff
@@ -180,7 +180,7 @@ public function callAggregateAllPlugins($visits, $visitsConverted, $forceArchivi
                         $this->params->getSegment() ? sprintf("(for segment = '%s')", $this->params->getSegment()->getString()) : ''
                     );
                 } catch (Exception $e) {
-                    throw new PluginsArchiverException($e->getMessage() . " - in plugin $pluginName with trace: " . $e->getTraceAsString(), $e->getCode(), $e);
```
Comment (tsteur) [inline core/ArchiveProcessor/PluginsArchiver.php:183]: @diosmosis could we use the exception to text processor to add the trace if enabled?

Comment (diosmosis) [reply core/ArchiveProcessor/PluginsArchiver.php:183]: This just throws the exception, the caused by exception is part of the chain. It should be seen and printed out when handled later on.

**Review by sgiehl — APPROVED** [review]

File: core/ArchiveProcessor/PluginsArchiver.php (hunk 1 above)
Comment (tsteur) [reply core/ArchiveProcessor/PluginsArchiver.php:183]: OK. Be good to double check this works though so the trace of the original exception will be seen before merging.

Comment (diosmosis) [reply core/ArchiveProcessor/PluginsArchiver.php:183]: Verified that if this is set in the bootstrap.php file:

```
define('PIWIK_PRINT_ERROR_BACKTRACE', 1);
define('PIWIK_ARCHIVE_NO_TRUNCATE', 1);
```

this is the exception stack trace:

```
?module=API&method=CoreAdminHome.archiveReports&idSite=1&period=day&date=2020-11-05&format=json&trigger=archivephp. Response was '{"result":"error","message":"test exception - in plugin Actions. #0 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(206): Piwik\\ArchiveProcessor\\PluginsArchiver-&gt;callAggregateAllPlugins('110', '84', false) #1 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(147): Piwik\\ArchiveProcessor\\Loader-&gt;prepareAllPluginsArchive('110', '84') #2 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(96): Piwik\\ArchiveProcessor\\Loader-&gt;prepareArchiveImpl(false) #3 \/Users\/ user\/Projects\/matomo\/core\/Context.php(75): Piwik\\ArchiveProcessor\\Loader-&gt;Piwik\\ArchiveProcessor\\{closure}() #4 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(97): Piwik\\Context::changeIdSite(1, Object(Closure)) #5 \/Users\/ user\/Projects\/matomo\/plugins\/CoreAdminHome\/API.php(278): Piwik\\ArchiveProcessor\\Loader-&gt;prepareArchive(false) #6 [internal function]: Piwik\\Plugins\\CoreAdminHome\\API-&gt;archiveReports('1', Object(Piwik\\Period\\Day), '2020-11-05', false, false, false) #7 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(244): call_user_func_array(Array, Array) #8 \/Users\/ user\/Projects\/matomo\/core\/Context.php(28): Piwik\\API\\Proxy-&gt;Piwik\\API\\{closure}() #9 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(335): Piwik\\Context::executeWithQueryParameters(Array, Object(Closure)) #10 \/Users\/ user\/Projects\/matomo\/core\/API\/Request.php(266): Piwik\\API\\Proxy-&gt;call('\\\\Piwik\\\\Plugins\\\\...', 'archiveReports', Array) #11 \/Users\/ user\/Projects\/matomo\/plugins\/API\/Controller.php(46): Piwik\\API\\Request-&gt;process() #12 [internal function]: Piwik\\Plugins\\API\\Controller-&gt;index() #13 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(605): call_user_func_array(Array, Array) #14 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(166): Piwik\\FrontController-&gt;doDispatch('API', false, Array) #15 \/Users\/ user\/Projects\/matomo\/core\/dispatch.php(32): Piwik\\FrontController-&gt;dispatch() #16 \/Users\/ user\/Projects\/matomo\/index.php(25): require_once('\/Users\/ user\/Pr...') #17 \/Users\/ user\/Projects\/matomo\/core\/CliMulti\/RequestCommand.php(79): require_once('\/Users\/ user\/Pr...') #18 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Command\/Command.php(257): Piwik\\CliMulti\\RequestCommand-&gt;execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #19 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(874): Symfony\\Component\\Console\\Command\\Command-&gt;run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #20 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(195): Symfony\\Component\\Console\\Application-&gt;doRunCommand(Object(Piwik\\CliMulti\\RequestCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #21 [internal function]: Symfony\\Component\\Console\\Application-&gt;doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #22 \/Users\/ user\/Projects\/matomo\/core\/Console.php(130): call_user_func(Array, Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #23 \/Users\/ user\/Projects\/matomo\/core\/Access.php(673): Piwik\\Console-&gt;Piwik\\{closure}() #24 \/Users\/ user\/Projects\/matomo\/core\/Console.php(131): Piwik\\Access::doAsSuperUser(Object(Closure)) #25 \/Users\/ user\/Projects\/matomo\/core\/Console.php(82): Piwik\\Console-&gt;doRunImpl(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #26 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(126): Piwik\\Console-&gt;doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #27 \/Users\/ user\/Projects\/matomo\/console(32): Symfony\\Component\\Console\\Application-&gt;run() #28 {main}, caused by: test exception #0 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(206): Piwik\\ArchiveProcessor\\PluginsArchiver-&gt;callAggregateAllPlugins('110', '84', false) #1 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(147): Piwik\\ArchiveProcessor\\Loader-&gt;prepareAllPluginsArchive('110', '84') #2 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(96): Piwik\\ArchiveProcessor\\Loader-&gt;prepareArchiveImpl(false) #3 \/Users\/ user\/Projects\/matomo\/core\/Context.php(75): Piwik\\ArchiveProcessor\\Loader-&gt;Piwik\\ArchiveProcessor\\{closure}() #4 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(97): Piwik\\Context::changeIdSite(1, Object(Closure)) #5 \/Users\/ user\/Projects\/matomo\/plugins\/CoreAdminHome\/API.php(278): Piwik\\ArchiveProcessor\\Loader-&gt;prepareArchive(false) #6 [internal function]: Piwik\\Plugins\\CoreAdminHome\\API-&gt;archiveReports('1', Object(Piwik\\Period\\Day), '2020-11-05', false, false, false) #7 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(244): call_user_func_array(Array, Array) #8 \/Users\/ user\/Projects\/matomo\/core\/Context.php(28): Piwik\\API\\Proxy-&gt;Piwik\\API\\{closure}() #9 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(335): Piwik\\Context::executeWithQueryParameters(Array, Object(Closure)) #10 \/Users\/ user\/Projects\/matomo\/core\/API\/Request.php(266): Piwik\\API\\Proxy-&gt;call('\\\\Piwik\\\\Plugins\\\\...', 'archiveReports', Array) #11 \/Users\/ user\/Projects\/matomo\/plugins\/API\/Controller.php(46): Piwik\\API\\Request-&gt;process() #12 [internal function]: Piwik\\Plugins\\API\\Controller-&gt;index() #13 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(605): call_user_func_array(Array, Array) #14 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(166): Piwik\\FrontController-&gt;doDispatch('API', false, Array) #15 \/Users\/ user\/Projects\/matomo\/core\/dispatch.php(32): Piwik\\FrontController-&gt;dispatch() #16 \/Users\/ user\/Projects\/matomo\/index.php(25): require_once('\/Users\/ user\/Pr...') #17 \/Users\/ user\/Projects\/matomo\/core\/CliMulti\/RequestCommand.php(79): require_once('\/Users\/ user\/Pr...') #18 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Command\/Command.php(257): Piwik\\CliMulti\\RequestCommand-&gt;execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #19 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(874): Symfony\\Component\\Console\\Command\\Command-&gt;run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #20 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(195): Symfony\\Component\\Console\\Application-&gt;doRunCommand(Object(Piwik\\CliMulti\\RequestCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #21 [internal function]: Symfony\\Component\\Console\\Application-&gt;doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #22 \/Users\/ user\/Projects\/matomo\/core\/Console.php(130): call_user_func(Array, Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #23 \/Users\/ user\/Projects\/matomo\/core\/Access.php(673): Piwik\\Console-&gt;Piwik\\{closure}() #24 \/Users\/ user\/Projects\/matomo\/core\/Console.php(131): Piwik\\Access::doAsSuperUser(Object(Closure)) #25 \/Users\/ user\/Projects\/matomo\/core\/Console.php(82): Piwik\\Console-&gt;doRunImpl(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #26 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(126): Piwik\\Console-&gt;doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #27 \/Users\/ user\/Projects\/matomo\/console(32): Symfony\\Component\\Console\\Application-&gt;run() #28 {main}, caused by: inner #0 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(206): Piwik\\ArchiveProcessor\\PluginsArchiver-&gt;callAggregateAllPlugins('110', '84', false) #1 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(147): Piwik\\ArchiveProcessor\\Loader-&gt;prepareAllPluginsArchive('110', '84') #2 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(96): Piwik\\ArchiveProcessor\\Loader-&gt;prepareArchiveImpl(false) #3 \/Users\/ user\/Projects\/matomo\/core\/Context.php(75): Piwik\\ArchiveProcessor\\Loader-&gt;Piwik\\ArchiveProcessor\\{closure}() #4 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(97): Piwik\\Context::changeIdSite(1, Object(Closure)) #5 \/Users\/ user\/Projects\/matomo\/plugins\/CoreAdminHome\/API.php(278): Piwik\\ArchiveProcessor\\Loader-&gt;prepareArchive(false) #6 [internal function]: Piwik\\Plugins\\CoreAdminHome\\API-&gt;archiveReports('1', Object(Piwik\\Period\\Day), '2020-11-05', false, false, false) #7 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(244): call_user_func_array(Array, Array) #8 \/Users\/ user\/Projects\/matomo\/core\/Context.php(28): Piwik\\API\\Proxy-&gt;Piwik\\API\\{closure}() #9 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(335): Piwik\\Context::executeWithQueryParameters(Array, Object(Closure)) #10 \/Users\/ user\/Projects\/matomo\/core\/API\/Request.php(266): Piwik\\API\\Proxy-&gt;call('\\\\Piwik\\\\Plugins\\\\...', 'archiveReports', Array) #11 \/Users\/ user\/Projects\/matomo\/plugins\/API\/Controller.php(46): Piwik\\API\\Request-&gt;process() #12 [internal function]: Piwik\\Plugins\\API\\Controller-&gt;index() #13 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(605): call_user_func_array(Array, Array) #14 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(166): Piwik\\FrontController-&gt;doDispatch('API', false, Array) #15 \/Users\/ user\/Projects\/matomo\/core\/dispatch.php(32): Piwik\\FrontController-&gt;dispatch() #16 \/Users\/ user\/Projects\/matomo\/index.php(25): require_once('\/Users\/ user\/Pr...') #17 \/Users\/ user\/Projects\/matomo\/core\/CliMulti\/RequestCommand.php(79): require_once('\/Users\/ user\/Pr...') #18 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Command\/Command.php(257): Piwik\\CliMulti\\RequestCommand-&gt;execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #19 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(874): Symfony\\Component\\Console\\Command\\Command-&gt;run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #20 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(195): Symfony\\Component\\Console\\Application-&gt;doRunCommand(Object(Piwik\\CliMulti\\RequestCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #21 [internal function]: Symfony\\Component\\Console\\Application-&gt;doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #22 \/Users\/ user\/Projects\/matomo\/core\/Console.php(130): call_user_func(Array, Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #23 \/Users\/ user\/Projects\/matomo\/core\/Access.php(673): Piwik\\Console-&gt;Piwik\\{closure}() #24 \/Users\/ user\/Projects\/matomo\/core\/Console.php(131): Piwik\\Access::doAsSuperUser(Object(Closure)) #25 \/Users\/ user\/Projects\/matomo\/core\/Console.php(82): Piwik\\Console-&gt;doRunImpl(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #26 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(126): Piwik\\Console-&gt;doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput)) #27 \/Users\/ user\/Projects\/matomo\/console(32): Symfony\\Component\\Console\\Application-&gt;run() #28 {main}","backtrace":"test exception - in plugin Actions.\n#0 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(206): Piwik\\ArchiveProcessor\\PluginsArchiver->callAggregateAllPlugins('110', '84', false)\n#1 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(147): Piwik\\ArchiveProcessor\\Loader->prepareAllPluginsArchive('110', '84')\n#2 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(96): Piwik\\ArchiveProcessor\\Loader->prepareArchiveImpl(false)\n#3 \/Users\/ user\/Projects\/matomo\/core\/Context.php(75): Piwik\\ArchiveProcessor\\Loader->Piwik\\ArchiveProcessor\\{closure}()\n#4 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(97): Piwik\\Context::changeIdSite(1, Object(Closure))\n#5 \/Users\/ user\/Projects\/matomo\/plugins\/CoreAdminHome\/API.php(278): Piwik\\ArchiveProcessor\\Loader->prepareArchive(false)\n#6 [internal function]: Piwik\\Plugins\\CoreAdminHome\\API->archiveReports('1', Object(Piwik\\Period\\Day), '2020-11-05', false, false, false)\n#7 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(244): call_user_func_array(Array, Array)\n#8 \/Users\/ user\/Projects\/matomo\/core\/Context.php(28): Piwik\\API\\Proxy->Piwik\\API\\{closure}()\n#9 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(335): Piwik\\Context::executeWithQueryParameters(Array, Object(Closure))\n#10 \/Users\/ user\/Projects\/matomo\/core\/API\/Request.php(266): Piwik\\API\\Proxy->call('\\\\Piwik\\\\Plugins\\\\...', 'archiveReports', Array)\n#11 \/Users\/ user\/Projects\/matomo\/plugins\/API\/Controller.php(46): Piwik\\API\\Request->process()\n#12 [internal function]: Piwik\\Plugins\\API\\Controller->index()\n#13 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(605): call_user_func_array(Array, Array)\n#14 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(166): Piwik\\FrontController->doDispatch('API', false, Array)\n#15 \/Users\/ user\/Projects\/matomo\/core\/dispatch.php(32): Piwik\\FrontController->dispatch()\n#16 \/Users\/ user\/Projects\/matomo\/index.php(25): require_once('\/Users\/ user\/Pr...')\n#17 \/Users\/ user\/Projects\/matomo\/core\/CliMulti\/RequestCommand.php(79): require_once('\/Users\/ user\/Pr...')\n#18 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Command\/Command.php(257): Piwik\\CliMulti\\RequestCommand->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#19 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(874): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#20 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(195): Symfony\\Component\\Console\\Application->doRunCommand(Object(Piwik\\CliMulti\\RequestCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#21 [internal function]: Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#22 \/Users\/ user\/Projects\/matomo\/core\/Console.php(130): call_user_func(Array, Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#23 \/Users\/ user\/Projects\/matomo\/core\/Access.php(673): Piwik\\Console->Piwik\\{closure}()\n#24 \/Users\/ user\/Projects\/matomo\/core\/Console.php(131): Piwik\\Access::doAsSuperUser(Object(Closure))\n#25 \/Users\/ user\/Projects\/matomo\/core\/Console.php(82): Piwik\\Console->doRunImpl(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#26 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(126): Piwik\\Console->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#27 \/Users\/ user\/Projects\/matomo\/console(32): Symfony\\Component\\Console\\Application->run()\n#28 {main},\ncaused by: test exception\n#0 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(206): Piwik\\ArchiveProcessor\\PluginsArchiver->callAggregateAllPlugins('110', '84', false)\n#1 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(147): Piwik\\ArchiveProcessor\\Loader->prepareAllPluginsArchive('110', '84')\n#2 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(96): Piwik\\ArchiveProcessor\\Loader->prepareArchiveImpl(false)\n#3 \/Users\/ user\/Projects\/matomo\/core\/Context.php(75): Piwik\\ArchiveProcessor\\Loader->Piwik\\ArchiveProcessor\\{closure}()\n#4 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(97): Piwik\\Context::changeIdSite(1, Object(Closure))\n#5 \/Users\/ user\/Projects\/matomo\/plugins\/CoreAdminHome\/API.php(278): Piwik\\ArchiveProcessor\\Loader->prepareArchive(false)\n#6 [internal function]: Piwik\\Plugins\\CoreAdminHome\\API->archiveReports('1', Object(Piwik\\Period\\Day), '2020-11-05', false, false, false)\n#7 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(244): call_user_func_array(Array, Array)\n#8 \/Users\/ user\/Projects\/matomo\/core\/Context.php(28): Piwik\\API\\Proxy->Piwik\\API\\{closure}()\n#9 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(335): Piwik\\Context::executeWithQueryParameters(Array, Object(Closure))\n#10 \/Users\/ user\/Projects\/matomo\/core\/API\/Request.php(266): Piwik\\API\\Proxy->call('\\\\Piwik\\\\Plugins\\\\...', 'archiveReports', Array)\n#11 \/Users\/ user\/Projects\/matomo\/plugins\/API\/Controller.php(46): Piwik\\API\\Request->process()\n#12 [internal function]: Piwik\\Plugins\\API\\Controller->index()\n#13 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(605): call_user_func_array(Array, Array)\n#14 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(166): Piwik\\FrontController->doDispatch('API', false, Array)\n#15 \/Users\/ user\/Projects\/matomo\/core\/dispatch.php(32): Piwik\\FrontController->dispatch()\n#16 \/Users\/ user\/Projects\/matomo\/index.php(25): require_once('\/Users\/ user\/Pr...')\n#17 \/Users\/ user\/Projects\/matomo\/core\/CliMulti\/RequestCommand.php(79): require_once('\/Users\/ user\/Pr...')\n#18 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Command\/Command.php(257): Piwik\\CliMulti\\RequestCommand->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#19 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(874): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#20 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(195): Symfony\\Component\\Console\\Application->doRunCommand(Object(Piwik\\CliMulti\\RequestCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#21 [internal function]: Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#22 \/Users\/ user\/Projects\/matomo\/core\/Console.php(130): call_user_func(Array, Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#23 \/Users\/ user\/Projects\/matomo\/core\/Access.php(673): Piwik\\Console->Piwik\\{closure}()\n#24 \/Users\/ user\/Projects\/matomo\/core\/Console.php(131): Piwik\\Access::doAsSuperUser(Object(Closure))\n#25 \/Users\/ user\/Projects\/matomo\/core\/Console.php(82): Piwik\\Console->doRunImpl(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#26 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(126): Piwik\\Console->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#27 \/Users\/ user\/Projects\/matomo\/console(32): Symfony\\Component\\Console\\Application->run()\n#28 {main},\ncaused by: inner\n#0 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(206): Piwik\\ArchiveProcessor\\PluginsArchiver->callAggregateAllPlugins('110', '84', false)\n#1 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(147): Piwik\\ArchiveProcessor\\Loader->prepareAllPluginsArchive('110', '84')\n#2 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(96): Piwik\\ArchiveProcessor\\Loader->prepareArchiveImpl(false)\n#3 \/Users\/ user\/Projects\/matomo\/core\/Context.php(75): Piwik\\ArchiveProcessor\\Loader->Piwik\\ArchiveProcessor\\{closure}()\n#4 \/Users\/ user\/Projects\/matomo\/core\/ArchiveProcessor\/Loader.php(97): Piwik\\Context::changeIdSite(1, Object(Closure))\n#5 \/Users\/ user\/Projects\/matomo\/plugins\/CoreAdminHome\/API.php(278): Piwik\\ArchiveProcessor\\Loader->prepareArchive(false)\n#6 [internal function]: Piwik\\Plugins\\CoreAdminHome\\API->archiveReports('1', Object(Piwik\\Period\\Day), '2020-11-05', false, false, false)\n#7 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(244): call_user_func_array(Array, Array)\n#8 \/Users\/ user\/Projects\/matomo\/core\/Context.php(28): Piwik\\API\\Proxy->Piwik\\API\\{closure}()\n#9 \/Users\/ user\/Projects\/matomo\/core\/API\/Proxy.php(335): Piwik\\Context::executeWithQueryParameters(Array, Object(Closure))\n#10 \/Users\/ user\/Projects\/matomo\/core\/API\/Request.php(266): Piwik\\API\\Proxy->call('\\\\Piwik\\\\Plugins\\\\...', 'archiveReports', Array)\n#11 \/Users\/ user\/Projects\/matomo\/plugins\/API\/Controller.php(46): Piwik\\API\\Request->process()\n#12 [internal function]: Piwik\\Plugins\\API\\Controller->index()\n#13 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(605): call_user_func_array(Array, Array)\n#14 \/Users\/ user\/Projects\/matomo\/core\/FrontController.php(166): Piwik\\FrontController->doDispatch('API', false, Array)\n#15 \/Users\/ user\/Projects\/matomo\/core\/dispatch.php(32): Piwik\\FrontController->dispatch()\n#16 \/Users\/ user\/Projects\/matomo\/index.php(25): require_once('\/Users\/ user\/Pr...')\n#17 \/Users\/ user\/Projects\/matomo\/core\/CliMulti\/RequestCommand.php(79): require_once('\/Users\/ user\/Pr...')\n#18 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Command\/Command.php(257): Piwik\\CliMulti\\RequestCommand->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#19 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(874): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#20 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(195): Symfony\\Component\\Console\\Application->doRunCommand(Object(Piwik\\CliMulti\\RequestCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#21 [internal function]: Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#22 \/Users\/ user\/Projects\/matomo\/core\/Console.php(130): call_user_func(Array, Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#23 \/Users\/ user\/Projects\/matomo\/core\/Access.php(673): Piwik\\Console->Piwik\\{closure}()\n#24 \/Users\/ user\/Projects\/matomo\/core\/Console.php(131): Piwik\\Access::doAsSuperUser(Object(Closure))\n#25 \/Users\/ user\/Projects\/matomo\/core\/Console.php(82): Piwik\\Console->doRunImpl(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#26 \/Users\/ user\/Projects\/matomo\/vendor\/symfony\/console\/Symfony\/Component\/Console\/Application.php(126): Piwik\\Console->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))\n#27 \/Users\/ user\/Projects\/matomo\/console(32): Symfony\\Component\\Console\\Application->run()\n#28 {main}"}
```

(notice the `caused by` bits)

**Comment by tsteur** [comment]
LGTM. Will need to resolve merge conflict though

**Pull request merged**